### PR TITLE
  Add new version of API with object store, sku selection and arm lite user

### PR DESCRIPTION
**Pull request by anponnet**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by anponnet** [comment]
it is approved by ARM team in private repo https://github.com/Azure/azure-rest-api-specs-pr/pull/967

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/databoxedge.json [hunk 1]
```diff
@@ -0,0 +1,4919 @@
...
+            "description": "The container Name",
+            "required": true,
+            "type": "string"
+          },
+          { "$ref": "#/parameters/subscriptionIdParameter" },
+          { "$ref": "#/parameters/resourceGroupNameParameter" },
+          { "$ref": "#/parameters/apiVersionParameter" }
+        ],
+        "responses": {
+          "20
```
Comment (ryansbenson) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/databoxedge.json]: size?

Comment (ryansbenson) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/databoxedge.json]: Can this be modeled? From an SDK/Documentation perspective this should be more customer focused.

Comment (ryansbenson) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/databoxedge.json]: readonly?

**Comment by anponnet** [comment]
@yungezz, @ryansbenson : Can you please complete the review and help to merge it. The same PR is reviewed and approved by ARM team in private repo https://github.com/Azure/azure-rest-api-specs-pr/pull/967

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/databoxedge.json [hunk 2]
```diff
@@ -3997,6 +3913,82 @@
...
+          "enum": [ "Standard" ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "SkuTier",
+            "modelAsString": true
+          }
+        },
+        "size": {
+          "description": "The Sku kind",
```
Comment (yungezz) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/databoxedge.json]: typo in description?

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/databoxedge.json [hunk 3]
```diff
@@ -3997,6 +3913,82 @@
...
+        "resourceType": {
+          "description": "The type of the resource",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "The Sku name",
+          "enum": [ "Gateway", "Edge", "TEA_1Node", "TEA_1Node_UPS", "TEA_1Node_Heater", "TEA_1Node_UPS_Heater", "TEA_4Node_Heater", "TEA_4Node_UPS_Heater", "TMA" ],
+          "type": "string",
+          "readOnly": true,
```
Comment (yungezz) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/databoxedge.json]: so user cannot specify sku of the resource, right?

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/examples/ListSkus.json [hunk 4]
```diff
@@ -1,138 +1,51 @@
...
+            "locations": [
+              "West US"
+            ],
+            "locationInfo": [
+              {
+                "location": "West US"
+              }
+            ],
+            "apiVersions": [
+              "2019-09-01"
```
Comment (yungezz) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/examples/ListSkus.json]: 2019-08-01?

**Comment by yungezz** [comment]
hi @anponnet  pls fix SpellCheck error from CI. thanks.

**Comment by anponnet** [comment]
Hi @yungezz I have fixed all comments except on spellcheck. The flagged contract was already approved (And it is a valid word) and it is in production, which we cannot change. Can you please add an exception and merge it?

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/examples/ListSkus.json [hunk 5]
```diff
@@ -1,4 +1,4 @@
-{
+2019-09-01{
```
Comment (yungezz) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/stable/2019-08-01/examples/ListSkus.json]: pls remove this. this cause CI failure

**Comment by yungezz** [comment]
> Hi @yungezz I have fixed all comments except on spellcheck. The flagged contract was already approved (And it is a valid word) and it is in production, which we cannot change. Can you please add an exception and merge it?

here's guide on how to fix SpellCheck error: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md#spell-check

**Review by anponnet — COMMENTED** [review]
> > Hi @yungezz I have fixed all comments except on spellcheck. The flagged contract was already approved (And it is a valid word) and it is in production, which we cannot change. Can you please add an exception and merge it?
> 
> here's guide on how to fix SpellCheck error: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md#spell-check

@yungezz , Thank you, I have fixed the errors and 17 checks are successful (One is marked as ignore-testonly)

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
HI @ryansbenson  could you pls have a look again?  The PR is reviewed in private repo: https://github.com/Azure/azure-rest-api-specs-pr/pull/967/files

**Comment by anponnet** [comment]
@yungezz , can you hold the merge till we confirm

**Comment by yungezz** [comment]
> @yungezz , can you hold the merge till we confirm

sure. Pls add label "DoNotMerge" and "[Do Not Merge]" in title to PR not ready for merging in future.

**Review by KrisBash — APPROVED** [review]
Signing off for ARM

**Comment by anponnet** [comment]
@yungezz I have made a one line change to API, it is a tag/operationId change... Can you see it once

**Comment by yungezz** [comment]
> @yungezz I have made a one line change to API, it is a tag/operationId change... Can you see it once

hi @anponnet  operationId change is breaking change, it will break SDKs generated from the swagger file. Does the changed operationId align with service behavior on production now?

**Review by anponnet — COMMENTED** [review]
> > @yungezz I have made a one line change to API, it is a tag/operationId change... Can you see it once
> 
> hi @anponnet operationId change is breaking change, it will break SDKs generated from the swagger file. Does the changed operationId align with service behavior on production now?

Hi @yungezz It is a new API and there is no SDK for this, so no breaking. I dont think there is any hard dependency between service code and operationId, it just maps sdk method to API in swagger, and makes http request with the url, is it right understanding?

**Comment by anponnet** [comment]
@yungezz , Our service deployment is complete, can you please merge this PR?

**Pull request merged**